lib/rpmsg/rpmsg_virtio.c

```diff
@@ -140,15 +140,15 @@ static int rpmsg_virtio_enqueue_buffer(struct rpmsg_virtio_device *rvdev,
  * return - pointer to buffer.
  */
 static void *rpmsg_virtio_get_tx_buffer(struct rpmsg_virtio_device *rvdev,
-					unsigned long *len,
+					unsigned int *len,
 					unsigned short *idx)
 {
 	unsigned int role = rpmsg_virtio_get_role(rvdev);
 	void *data = NULL;
 
 #ifndef VIRTIO_SLAVE_ONLY
 	if (role == RPMSG_MASTER) {
-		data = virtqueue_get_buffer(rvdev->svq, (uint32_t *)len, idx);
+		data = virtqueue_get_buffer(rvdev->svq, len, idx);
```

Comment:
I have got compilation error with this patch on cortexr5 with gnu gcc toolchain:

ws/jliang/source/open-amp/lib/rpmsg/rpmsg_virtio.c: In function 'rpmsg_virtio_get_tx_buffer':
/ws/jliang/source/open-amp/lib/rpmsg/rpmsg_virtio.c:146:43: error: passing argument 2 of 'virtqueue_get_buffer' from incompatible pointer type [-Werror=incompatible-pointer-types]
   data = virtqueue_get_buffer(rvdev->svq, len, idx);
                                           ^~~
In file included from /ws/jliang/source/open-amp/lib/include/openamp/virtio.h:10:0,
                 from /ws/jliang/source/open-amp/lib/include/openamp/rpmsg_virtio.h:18,
                 from /ws/jliang/source/open-amp/lib/rpmsg/rpmsg_virtio.c:13:
/ws/jliang/source/open-amp/lib/include/openamp/virtqueue.h:191:7: note: expected 'uint32_t * {aka long unsigned int *}' but argument is of type 'unsigned int *'
 void *virtqueue_get_buffer(struct virtqueue *vq, uint32_t *len, uint16_t *idx);
       ^~~~~~~~~~~~~~~~~~~~
/ws/jliang/source/open-amp/lib/rpmsg/rpmsg_virtio.c:157:58: error: passing argument 3 of 'virtqueue_get_available_buffer' from incompatible pointer type [-Werror=incompatible-pointer-types]
   data = virtqueue_get_available_buffer(rvdev->svq, idx, len);
                                                          ^~~
In file included from /ws/jliang/source/open-amp/lib/include/openamp/virtio.h:10:0,
                 from /ws/jliang/source/open-amp/lib/include/openamp/rpmsg_virtio.h:18,
                 from /ws/jliang/source/open-amp/lib/rpmsg/rpmsg_virtio.c:13:
/ws/jliang/source/open-amp/lib/include/openamp/virtqueue.h:193:7: note: expected 'uint32_t * {aka long unsigned int *}' but argument is of type 'unsigned int *'
 void *virtqueue_get_available_buffer(struct virtqueue *vq, uint16_t *avail_idx,
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/ws/jliang/source/open-amp/lib/rpmsg/rpmsg_virtio.c: In function 'rpmsg_virtio_get_rx_buffer':
/ws/jliang/source/open-amp/lib/rpmsg/rpmsg_virtio.c:185:43: error: passing argument 2 of 'virtqueue_get_buffer' from incompatible pointer type [-Werror=incompatible-pointer-types]
   data = virtqueue_get_buffer(rvdev->rvq, len, idx);
                                           ^~~
In file included from /ws/jliang/source/open-amp/lib/include/openamp/virtio.h:10:0,
                 from /ws/jliang/source/open-amp/lib/include/openamp/rpmsg_virtio.h:18,
                 from /ws/jliang/source/open-amp/lib/rpmsg/rpmsg_virtio.c:13:
/ws/jliang/source/open-amp/lib/include/openamp/virtqueue.h:191:7: note: expected 'uint32_t * {aka long unsigned int *}' but argument is of type 'unsigned int *'
 void *virtqueue_get_buffer(struct virtqueue *vq, uint32_t *len, uint16_t *idx);
       ^~~~~~~~~~~~~~~~~~~~
/ws/jliang/source/open-amp/lib/rpmsg/rpmsg_virtio.c:192:55: error: passing argument 3 of 'virtqueue_get_available_buffer' from incompatible pointer type [-Werror=incompatible-pointer-types]
       virtqueue_get_available_buffer(rvdev->rvq, idx, len);
                                                       ^~~
In file included from /ws/jliang/source/open-amp/lib/include/openamp/virtio.h:10:0,
                 from /ws/jliang/source/open-amp/lib/include/openamp/rpmsg_virtio.h:18,
                 from /ws/jliang/source/open-amp/lib/rpmsg/rpmsg_virtio.c:13:
/ws/jliang/source/open-amp/lib/include/openamp/virtqueue.h:193:7: note: expected 'uint32_t * {aka long unsigned int *}' but argument is of type 'unsigned int *'
 void *virtqueue_get_available_buffer(struct virtqueue *vq, uint16_t *avail_idx,
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

As we use uint32_t in the virtqueue.h, will need to make sure the input arg to this function is uint32_t.

Reply:
Ok, I will change the type to uint32_t*.